### PR TITLE
Use dedicated permalink creators in search results with multiple rooms

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/9376

This also would have happened in "All Rooms" searches.

### CODE REVIEW

**Comment by turt2live** [comment]
> That said, the start/stop behaviour is different between the two and I'm not sure why.

It's because for some reason I thought start was called in the constructor, but it isn't, so the confusion isn't required. Will remove.

File: src/components/structures/RoomView.js [hunk 1]
```diff
@@ -650,11 +662,9 @@ module.exports = React.createClass({
...
         this._calculateRecommendedVersion(room);
         this._updateE2EStatus(room);
-        if (!this.state.permalinkCreator) {
-            const permalinkCreator = new RoomPermalinkCreator(room);
-            permalinkCreator.start();
-            this.setState({permalinkCreator});
-        }
+
+        let creator = this._getPermalinkCreatorForRoom(room);
+        if (!creator.isStarted()) creator.start();
```
Comment (bwindels) [inline src/components/structures/RoomView.js]: hm, I'm slightly worried if a search result spans lots of rooms, that we'd create listeners for all of those rooms. As the search view is not live, we're also not really interested in observing the most common servers.Also, as it stands, we'd also not stop listening on any of the room until we go to a different room.

I realize this is sort of opposite to what Dave said, but you could call just `load` and not `start` for the link creators created for the search tiles, and only call start on the one created for the current room... e.g. similar to what you had before :see_no_evil:

Search should really be in it's own component :cry:

Comment (turt2live) [reply src/components/structures/RoomView.js]: github ate my comments it seems :(

> Search should really be in it's own component 😢

+10000

> hm, I'm slightly worried if a search result spans lots of rooms, that we'd create listeners for all of those rooms. As the search view is not live, we're also not really interested in observing the most common servers.Also, as it stands, we'd also not stop listening on any of the room until we go to a different room.
> 
> I realize this is sort of opposite to what Dave said, but you could call just load and not start for the link creators created for the search tiles, and only call start on the one created for the current room... e.g. similar to what you had before 🙈

The permalink creator for the primary room should be the only that is started, the others are just created. They weren't being loaded though, which is fixed in https://github.com/matrix-org/matrix-react-sdk/pull/2898/commits/ccf292f05368f63dc4f0f5fd01135224d38f5b97 .

**Comment by turt2live** [comment]
> I'm also thinking that putting the search permalink creators and the one for the current room together in one collection is further entangling the room and search code, but this is not a huge deal.

I think it's better to have them merged for now as it reduces the number of variables going through that component. I do think that if we ever touch search again that we should split it out.

----

I've also made the load function either start or load the creator to save on calls.

**Pull request merged**